tests/test_1072-test-v2-sort.py

```diff
@@ -0,0 +1,802 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
+import pytest  # noqa: F401
+import numpy as np  # noqa: F401
+import awkward as ak  # noqa: F401
+
+from awkward._v2.tmp_for_testing import v1_to_v2, v2_to_v1
+
+pytestmark = pytest.mark.skipif(
+    ak._util.py27, reason="No Python 2.7 support in Awkward 2.x"
+)
+
+
+def test_keep_None_in_place_test():
+    v1_array = ak.Array([[3, 2, 1], [], None, [4, 5]])
+
+    assert ak.to_list(ak.argsort(v1_array, axis=1)) == [
+        [2, 1, 0],
+        [],
+        None,
+        [0, 1],
+    ]
+
+    assert ak.to_list(ak.sort(v1_array, axis=1)) == [
+        [1, 2, 3],
+        [],
+        None,
+        [4, 5],
+    ]
+
+    assert ak.to_list(v1_array[ak.argsort(v1_array, axis=1)]) == ak.to_list(
+        ak.sort(v1_array, axis=1)
+    )
+
+    v2_array = v1_to_v2(v1_array.layout)
+    assert ak.to_list(v2_array.sort(axis=1)) == ak.to_list(ak.sort(v1_array, axis=1))
+
+
+def test_empty_slice():
+    # muon = ak.Array([[{"pt": 1.0}], []], with_name="muon")
+    electron = ak.Array([[], [{"pt": 1.0}]], with_name="electron")
+
+    electron = electron[electron.pt > 5]
+    # FIXME: AttributeError: 'ListOffsetArray' object has no attribute 'pt'
+    # v2_electron = v1_to_v2(electron.layout)
+    # v2_electron = v2_electron[v2_electron.pt > 5]
+
+    assert ak.to_list(electron) == [[], []]
+    # assert ak.to_list(v2_electron) == [[], []]
+
+    id = ak.argsort(electron, axis=1)
+
+    assert ak.to_list(electron[id]) == [[], []]
+
+
+def test_masked():
+    v1_array = ak.Array([[0, 1, 2, 3], [3, 3, 3, 2, 1]])
+    is_valid = v1_array != 3
+
+    assert v1_array.mask[is_valid].tolist() == [
+        [0, 1, 2, None],
+        [None, None, None, 2, 1],
+    ]
+
+    assert ak.sort(v1_array.mask[is_valid]).tolist() == [
+        [0, 1, 2, None],
+        [1, 2, None, None, None],
+    ]
+
+    # FIXME: AttributeError: 'ListOffsetArray' object has no attribute 'mask'
+    # v2_array = v1_to_v2(v1_array.layout)
+    # assert v2_array.mask[is_valid].tolist() == [[0, 1, 2, None], [None, None, None, 2, 1]]
+    #
+    # assert ak.to_list(v2_array.mask[is_valid]) == [
+    #     [0, 1, 2, None],
+    #     [1, 2, None, None, None],
+    # ]
```

Comment:
So that we don't forget about it, I'm calling this out here: it looks like a slicing bug unrelated to sorting.

Reply:
BTW, I'm looking into an `argsort` now and in another test I'm running into the following `FIXME`:
```python
                     # FIXME: might infinite-loop before simplify_optiontype is implemented
     >           return prepare_tuple_nested(next.simplify_optiontype())
     E           UnboundLocalError: local variable 'next' referenced before assignment
    
     awkward/_v2/_slicing.py:218: UnboundLocalError

     assert ak.to_list(array[ak.argsort(array, axis=1)]) == ak.to_list(
         ak.sort(array, axis=1)
     )
```

It is `v2`, I've added the following to covert it - not sure if it's a good idea though: 

```python
diff --git a/src/awkward/operations/convert.py b/src/awkward/operations/convert.py
index 3a6983b3..17f73533 100644
--- a/src/awkward/operations/convert.py
+++ b/src/awkward/operations/convert.py
@@ -1920,6 +1920,10 @@ def to_layout(
     elif isinstance(array, Iterable):
         return from_iter(array, highlevel=False)
 
+    # FIXME?
+    elif isinstance(array, ak._v2.contents.Content):
+        return array
+
     elif not allow_other:
         raise TypeError(
             "{0} cannot be converted into an Awkward Array".format(array)
```